base/lib/stellar/sep10.rb

```diff
@@ -62,10 +73,11 @@ def self.build_challenge_tx(server:, client:, anchor_name:, timeout: 300)
     # = Example
     #
     #   sep10 = Stellar::SEP10
-    #   challenge = sep10.build_challenge_tx(server: server, client: user, anchor_name: anchor, timeout: timeout)
-    #   envelope, client_address = sep10.read_challenge_tx(challenge: challenge, server: server)
+    #   server = Stellar::KeyPair.random # this should be the SIGNING_KEY from your stellar.toml
+    #   challenge = sep10.build_challenge_tx(server: server, client: user, home_domain: domain, timeout: timeout)
+    #   envelope, client_address = sep10.read_challenge_tx(server: server, challenge: challenge)
     #
-    def self.read_challenge_tx(challenge_xdr:, server:)
+    def self.read_challenge_tx(server:, challenge_xdr:)
```

Comment:
Why is `home_domain` not added here? See the other SDK changes.

Reply:
This method is supposed to be used by a server in the implementation of the token endpoint (second phase of the protocol). The only parameter submitted to this endpoint by the client is encoded transaction. Server uses this method to decode the transaction, then it can extract the domain from the decoded envelope. 

We should probably even remove the `server` arg from this method, because in the scenario of single auth server handling multiple domains server needs to know the home domain for the specific challenge tx in order to identify the corresponding signing key. So this method needs to take the encoded transaction, verifies that it's structured as a challenge tx and then returns all the bits of decoded info in a convenient form.

Reply:
This function is used by clients too, to verify it meets the SEP-10 standard as well as get the decoded TransactionEnvelope. It should check the value of the manage data op, just like the other SDK’s.

Reply:
Actually I see why it might be nice to not check the value so ruby clients can read SEP-10 2.0 challenges